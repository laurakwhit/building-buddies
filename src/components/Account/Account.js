import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Account.scss';
import Nav from '../Nav/Nav';

class Account extends Component {
  render() {
    const { handleLogOut } = this.props;
    return (
      <div className="account">
        <Nav handleLogOut={handleLogOut} />
        <div className="main-content">
          <h1>Account</h1>
        </div>
      </div>
    );
  }
}

Account.propTypes = {
  handleLogOut: PropTypes.func.isRequired
};

export default Account;
