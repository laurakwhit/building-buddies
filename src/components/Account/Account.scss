.account {
  background-color: crimson;

  .main-content {
    overflow: scroll;
    width: calc(100% - 20rem);
  }
}
