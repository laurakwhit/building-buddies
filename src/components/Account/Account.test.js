import React from 'react';
import { shallow } from 'enzyme';

import Account from './Account';

describe('Account component', () => {
  let wrapper;
  let mockHandleLogOut;

  beforeEach(() => {
    mockHandleLogOut = jest.fn();
    wrapper = shallow(<Account handleLogOut={mockHandleLogOut} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
