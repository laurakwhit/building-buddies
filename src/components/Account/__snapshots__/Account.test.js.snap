// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Account
    handleLogOut={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav
          handleLogOut={[MockFunction]}
        />,
        <div
          className="main-content"
        >
          <h1>
            Account
          </h1>
        </div>,
      ],
      "className": "account",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleLogOut": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Account
          </h1>,
          "className": "main-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Account",
          },
          "ref": null,
          "rendered": "Account",
          "type": "h1",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav
            handleLogOut={[MockFunction]}
          />,
          <div
            className="main-content"
          >
            <h1>
              Account
            </h1>
          </div>,
        ],
        "className": "account",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleLogOut": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Account
            </h1>,
            "className": "main-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Account",
            },
            "ref": null,
            "rendered": "Account",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
